pkg/plugins/backendplugin/instrumentation/instrumentation.go

```diff
@@ -34,29 +42,51 @@ func instrumentPluginRequest(pluginID string, endpoint string, fn func() error)
 		status = "error"
 	}
 
-	elapsed := time.Since(start) / time.Millisecond
-	pluginRequestDuration.WithLabelValues(pluginID, endpoint).Observe(float64(elapsed))
-	pluginRequestCounter.WithLabelValues(pluginID, endpoint, status).Inc()
+	elapsed := time.Since(start)
+	pluginRequestDuration.WithLabelValues(pluginCtx.PluginID, endpoint).Observe(float64(elapsed / time.Millisecond))
+	pluginRequestCounter.WithLabelValues(pluginCtx.PluginID, endpoint, status).Inc()
+
+	if cfg.IsFeatureToggleEnabled(featuremgmt.FlagDatasourceLogger) {
+		logParams := []interface{}{
+			"status", status,
+			"duration", elapsed,
+			"plugin_id", pluginCtx.PluginID,
+			"endpoint", endpoint,
+		}
+
+		traceID := tracing.TraceIDFromContext(ctx, false)
+		if traceID != "" {
+			logParams = append(logParams, "traceID", traceID)
+		}
+
+		if pluginCtx.DataSourceInstanceSettings != nil {
+			logParams = append(logParams, "ds_name", pluginCtx.DataSourceInstanceSettings.Name)
+			logParams = append(logParams, "ds_uid", pluginCtx.DataSourceInstanceSettings.UID)
+			logParams = append(logParams, "ds_type", pluginCtx.DataSourceInstanceSettings.Type)
+		}
+
+		logger.Info("Plugin Request Completed", logParams...)
```

Comment:
Debug?

```suggestion
		logger.Debug("Plugin Request Completed", logParams...)
```

Reply:
Nah we want these logs for every plugin request in prod. turning on debug everywhere would be too much :)

This is the first stab at solving https://github.com/grafana/grafana/discussions/43120

Reply:
Expect more changes behind this feature flag :)

Reply:
Okay, but whenever you remove the feature toggle please change to debug or a setting ;)

Reply:
> a setting ;)

that would be fine. Similar to the request logger.